frontend/packages/ceph-storage-plugin/src/components/modals/storage-pool-modal/storage-pool-modal.tsx

```diff
@@ -0,0 +1,391 @@
+      apiVersion: apiVersionForModel(CephBlockPoolModel),
+      kind: CephBlockPoolModel.kind,
+      metadata: {
+        name: newPoolName,
+        namespace: CEPH_STORAGE_NAMESPACE,
+      },
+      spec: {
+        deviceClass: deviceClass || '',
+        parameters: {
+          compression_mode: isCompressed ? 'aggressive' : '', // eslint-disable-line @typescript-eslint/camelcase
+        },
+        replicated: {
+          size: Number(replicaSize),
+        },
+      },
+    };
+
+    handlePromise(k8sCreate(CephBlockPoolModel, poolObj))
+      .then(() => {
+        setIsSubmit(true);
+        // The modal will wait for 15 sec to get feedback from Rook
+        const timeoutTimer = setTimeout(() => {
+          setPoolStatus('timeout');
+          setIsSubmit(false);
+        }, 15000);
+        setTimer(timeoutTimer);
+      })
+      .catch(() => {
+        setPoolStatus('failed');
+      });
+  };
+
+  if (storageClusterLoading) {
+    return (
+      <EmptyState>
+        <EmptyStateIcon variant="container" component={LoadingComponent} />
+      </EmptyState>
+    );
+  }
+
+  if (poolStatus) {
+    return (
+      <div className="modal-content modal-content--no-inner-scroll" key="progress-modal">
+        <ModalTitle>{modalTitle}</ModalTitle>
+        <ModalBody>
+          <p>{modalDescription}</p>
+          <PoolStatusComponent status={poolStatus} name={newPoolName} error={errorMessage} />
+        </ModalBody>
+        <ModalFooter inProgress={poolStatus === 'progress'}>
+          <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
+            {poolStatus === 'failed' && (
+              <Button
+                type="button"
+                variant="secondary"
+                data-test-id="modal-cancel-action"
+                onClick={(e) => {
+                  e.preventDefault();
+                  setPoolStatus('');
+                }}
+              >
+                Try Again
+              </Button>
+            )}
+            <Button
+              type="submit"
+              variant="primary"
+              isDisabled={poolStatus === 'progress'}
+              id="confirm-action"
+              onClick={(e) => {
+                e.preventDefault();
+                if (poolStatus === 'created') {
+                  onPoolCreation(newPoolName);
+                }
+                close();
+              }}
+            >
+              Finish
+            </Button>
+          </ActionGroup>
+        </ModalFooter>
+      </div>
+    );
+  }
+
+  return (
+    <form
+      onSubmit={submit}
+      className="modal-content modal-content--no-inner-scroll"
+      key="pool-form-modal"
+    >
+      <ModalTitle>{modalTitle}</ModalTitle>
+      <ModalBody>
+        <p>{modalDescription}</p>
+        {!storageClusterLoading &&
+        !poolStatus &&
+        !storageClusterError &&
+        storageClusterReady &&
+        cephClusterObj[0]?.status?.phase === 'Ready' ? (
```

Comment:
Moving these conditions to a function and calling here, might be a better approach as so many conditions are checked.

Reply:
ack